internal/compiler/expression_tree.rs

```diff
@@ -1075,15 +1075,17 @@ impl Expression {
                 },
             };
             Expression::Cast { from: Box::new(from), to: target_type }
-        } else if matches!((&ty, &target_type, &self), (Type::Array(left), Type::Array(right), Expression::Array{..})
-            if left.can_convert(right) || **left == Type::Invalid)
```

Comment:
I'm curious, why does the can_convert check not work in the game example?

Reply:
Generally, one cannot, eg, `[int]` to `[string]` because they are not the same model. ( One would need a Map model)

The struct construction are not the same type because named struct.

But that work with array literal. Because we can do the conversion at compile time.

Reply:
Ahh, I see. Thanks!